packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+import { LivenessGuard } from "src/Safe/LivenessGuard.sol";
+
+// TODO(maurelian): remove me
+import { console2 as console } from "forge-std/console2.sol";
+
+/// @title LivenessModule
+/// @notice This module is intended to be used in conjunction with the LivenessGuard. It should be able to
+///         execute a transaction on the Safe in only a small number of cases:
+contract LivenessModule {
+    /// @notice The Safe contract instance
+    Safe public safe;
+
+    /// @notice The LivenessGuard contract instance
+    LivenessGuard public livenessGuard;
+
+    /// @notice The interval, in seconds, during which an owner must have demonstrated liveness
+    uint256 public livenessInterval;
+
+    /// @notice The minimum number of owners before ownership of the safe is transferred to the fallback owner.
+    uint256 public minOwners;
+
+    /// @notice The fallback owner of the Safe
+    address public fallbackOwner;
```

Comment:
Consider adding setter methods for these, or making them immutable. Probably simpler to make them immutable, and swap out the module if we decide to change the properties, since that limits the surface area of bugs in the module, which is nice because modules are very powerful

Reply:
Definitely +1 on making these immutable, ideally we can set no storage in the account, otherwise we will have headaches longer term

Reply:
Great callout. This contract now has only immutable state, and it feels great.